File: go/data_structures/stack.go
package stack

type Stack struct {
	items []string
}

func NewStack() *Stack {
	return &Stack{items: []string{}}
}

func (s *Stack) IsEmpty() bool {
	return len(s.items) == 0
}

func (s *Stack) lastIndex() int {
	return s.Size() - 1
}

func (s *Stack) Size() int {
	return len(s.items)
}

func (s *Stack) Push(item string) {
	s.items = append(s.items, item)
}

func (s *Stack) Pop() string {
	i := s.lastIndex()
	lastItem := s.items[i]
	s.items = s.items[:i]
	return lastItem
}

func (s *Stack) Peek() string {
	return s.items[s.lastIndex()]
}

func main() {
	stack := NewStack()
	stack.Push("apple")
	stack.Push("banana")
	stack.Push("orange")
	stack.Pop()
}

File: python/src/data_structures/stack.py
class Stack[T]:
    def __init__(self) -> None:
        self.items: list[T] = []

    def push(self, item: T) -> None:
        self.items.append(item)

    def pop(self) -> T:
        return self.items.pop()

    def peek(self) -> T:
        return self.items[-1]

    def size(self) -> int:
        return len(self.items)


if __name__ == "__main__":
    stack = Stack[str]()
    stack.push("apple")
    stack.push("banana")
    stack.push("orange")
    stack.pop()
